extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AutoInjectFieldProcessor.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.arc.deployment;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.processor.AnnotationsTransformer;
+import io.quarkus.arc.processor.DotNames;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+
+public class AutoInjectFieldProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(AutoInjectFieldProcessor.class);
+
+    @BuildStep
+    void autoInjectQualifiers(BeanArchiveIndexBuildItem beanArchiveIndex,
+            BuildProducer<AutoInjectAnnotationBuildItem> autoInjectAnnotations) {
+        List<DotName> qualifiers = new ArrayList<>();
+        for (AnnotationInstance qualifier : beanArchiveIndex.getIndex().getAnnotations(DotNames.QUALIFIER)) {
```

Comment:
Does this index automatically contain all the qualifier classes?

Reply:
It should. This index is built by `io.quarkus.arc.deployment.BeanArchiveProcessor`.

Reply:
Ah OK, I see we also index the annotations of the beans so the qualifier should come with that. Just wanted to be sure about that.